client/components/sub-masterbar-nav/index.jsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { findDOMNode } from 'react-dom';
+import classNames from 'classnames';
+
+import Gridicon from 'gridicons';
+import Item from './item';
+
+const VIEW_BREAKPOINT = 660;
+const VIEW_PADDING = 50;
+const ITEM_WIDTH = 110;
+
+export default class SubMasterbarNav extends Component {
+	static propTypes = {
+		fallback: PropTypes.shape(
+			{
+				label: PropTypes.string.isRequired,
+				uri: PropTypes.string.isRequired,
+				icon: PropTypes.string
+			}
+		),
+		options: PropTypes.arrayOf(
+			PropTypes.shape( {
+				label: PropTypes.string.isRequired,
+				uri: PropTypes.string.isRequired,
+				icon: PropTypes.string
+			} )
+		),
+		uri: PropTypes.string.isRequired
+	};
+
+	static defaultProps = {
+		options: []
+	}
+
+	state = {
+		collapsed: true
+	};
+
+	componentDidMount() {
+		this.onResize();
+
+		window.addEventListener( 'resize', this.onResize.bind( this ) );
+		window.addEventListener( 'scroll', this.onScroll.bind( this ) );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.onResize.bind( this ) );
+		window.removeEventListener( 'scroll', this.onScroll.bind( this ) );
+	}
+
+	render() {
+		const className = classNames(
+			'sub-masterbar-nav',
+			{ 'is-collapsed': this.state.collapsed }
+		);
+
+		const ellipsisClass = classNames(
+			'sub-masterbar-nav__ellipsis',
+			{ 'is-open': ! this.state.collapsed }
+		);
+
+		return (
+			<div className={ className }>
+				{ this.isDroppable( this.state.width ) &&
+					<div className="sub-masterbar-nav__select" onClick={ this.toggleList.bind( this ) }>
+						<Item
+							isSelected={ true }
+							label={ this.getSelected().label }
+							icon={ this.getSelected().icon }
+						/>
+						<Gridicon icon="chevron-down" className="sub-masterbar-nav__select-icon" />
+					</div>
+				}
+				<div className="sub-masterbar-nav__wrapper">
+					<div className="sub-masterbar-nav__items">
+						{ this.renderItems() }
+					</div>
+				</div>
+				{ this.isFoldable( this.state.width, this.props.options ) && (
+					<div className="sub-masterbar-nav__switch">
+						<Gridicon icon="ellipsis" className={ ellipsisClass } onClick={ this.toggleList.bind( this ) } />
+					</div>
+				)}
+			</div>
+		);
+	}
+
+	renderItems() {
+		return this.props.options.map( ( item, index ) =>
+			<Item
+				key={ index }
+				isSelected={ this.isSelected( item ) }
+				onClick={ this.selectItem.bind( this, item ) }
+				label={ item.label }
+				icon={ item.icon }
+				href={ item.uri }
+			/>
+		);
+	}
+
+	selectItem( option ) {
+		if ( ! option ) {
+			return;
+		}
+
+		this.setState( ( state ) => ( {
+			collapsed: this.isDroppable( state.width ) || state.collapsed
+		} ) );
+	}
+
+	toggleList() {
```

Comment:
Better use [this style](https://babeljs.io/blog/2015/06/07/react-on-es6-plus) (in case that's not obvious from [my other comment](https://github.com/Automattic/wp-calypso/pull/11903/files#r109200435) 😄 ):

```js
class PostInfo extends React.Component {
  handleOptionsButtonClick = (e) => {
    this.setState({showOptionsModal: true});
  }
}
```